src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -0,0 +1,57 @@
+package lotto.domain;
+
+import java.util.List;
+
+public class LottoNumber {
+
+    private final List<Integer> numbers;
+
+    public LottoNumber(Integer... numbers) {
+        this(List.of(numbers));
+    }
+
+    public LottoNumber(List<Integer> numbers) {
+        validation(numbers);
+        this.numbers = numbers;
+    }
+
+    public LottoResult checkLottoNumber(List<LottoNumber> useLottoNumber) {
+        LottoResult lottoResult = new LottoResult();
+
+        for (LottoNumber lottoNumber : useLottoNumber) {
+            int count = 0;
+
+            count = successCount(lottoNumber, count);
+
+            LottoRank lottoRank = LottoRank.fromMatchCount(count);
```

Comment:
```suggestion
            LottoRank lottoRank = lottoNumber.match(numbers);
```
이와 같이 접근해 보는 것은 어떨까?
LottoNumber에게 '너 몇등이야?'와 같은 메시지를 보내는 방식을 접근